### PR TITLE
Pipeline

### PR DESCRIPTION
**Pull request by dkm2110**
Added pipelining to the group communication code...The code still needs to be cleaned a bit in terms of log messages.

### CODE REVIEW

File: lang/cs/Org.Apache.REEF.Network/Group/CommonOperators/DefaultCommonOperatorNames.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Org.Apache.REEF.Network.Group.CommonOperators
```
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.Network/Group/CommonOperators/DefaultCommonOperatorNames.cs]: Could you please split default operators and pipeline changes into two separate PR? You need to create branches for each.

**Comment by jwang98052** [comment]
Branch name convention follows Jira name like REEF-101. You can create separate branches and merge your code over.

**Comment by jwang98052** [comment]
PR name convention starts from [REEF-XXX].... that follows the same JIRA number.

File: lang/cs/Org.Apache.REEF.Network/Group/CommonOperators/DefaultCommonOperatorNames.cs (hunk 1 above)
Comment (markusweimer) [inline lang/cs/Org.Apache.REEF.Network/Group/CommonOperators/DefaultCommonOperatorNames.cs]: Also, please provide a JIRA for these changes.

**Comment by markusweimer** [comment]
There are now two PRs open for this? Are they both meant to be looked at? If not, please close the one(s) that are not.

**Comment by dkm2110** [comment]
DO not need to review this code...Obsolete

**Comment by dkm2110** [comment]
Closed this one...please look at pipeliningonly

**Pull request closed without merging**